### PR TITLE
[ Ent-Corda ] Add ambassador certs creation role

### PR DESCRIPTION
**Pull request by jagpreetsinghsasan**
**Changelog**
- Add ambassador certs creation role

 

**Reviewed by**
@lakshyakumar 

 

**Linked issue**
#809

### CODE REVIEW

File: platforms/corda-ent/configuration/roles/create/certificates/ambassador/cenm-notary/tasks/nested_main.yaml [hunk 1]
```diff
@@ -0,0 +1,229 @@
...
+    path: "{{ ambassadortls }}"
+    check: "ensure_dir"
+
+############################################################################################
+###################### CHECK IF ROOT CERTIFICATES ARE PRESENT IN VAULT #####################
+
+# Checks if root certs are already created and stored in vault for CENM
+- name: Check if root certs are already created and stored in the vault for CENM
+  shell: |
+    vault kv get -format=yaml secret/cenm/{{ peer.name | lower }}/tlscerts
```
Comment (sownak) [inline platforms/corda-ent/configuration/roles/create/certificates/ambassador/cenm-notary/tasks/nested_main.yaml:36]: This is not the path for root certs. Root certs is at /secrets/cenm/root/certs

File: platforms/corda-ent/configuration/roles/create/certificates/ambassador/cenm-notary/tasks/nested_main.yaml [hunk 2]
```diff
@@ -0,0 +1,229 @@
...
+    VAULT_ADDR: "{{ org.vault.url }}"
+    VAULT_TOKEN: "{{ org.vault.root_token }}"
+  register: root_certs_cenm
+  ignore_errors: yes
+  when: org.type == 'cenm'
+
+# Checks if root certs are already created and stored in vault for notary
+- name: Check if root certs are already created and stored in vault for notary
+  shell: |
+    vault kv get -format=yaml secret/{{ peer.name }}/tlscerts
```
Comment (sownak) [inline platforms/corda-ent/configuration/roles/create/certificates/ambassador/cenm-notary/tasks/nested_main.yaml:47]: this path is not root cert, this is ambassador certs

File: platforms/corda-ent/configuration/roles/create/certificates/ambassador/cenm-notary/tasks/nested_main.yaml [hunk 3]
```diff
@@ -0,0 +1,229 @@
...
+- name: Store the existing key.jks file in a variable
+  stat:
+    path: "{{ rootca }}/rootca.jks"
+  register: rootca_stat_result
+
+############################################################################################
+#################### GENERATE ROOT CA CERTIFICATES, IF ABSENT IN VAULT #####################
+
+# Generate Root CA certificates
+- name: Generate Root CA certificates
```
Comment (sownak) [inline platforms/corda-ent/configuration/roles/create/certificates/ambassador/cenm-notary/tasks/nested_main.yaml:92]: Root CA certs are generated by PKI generator job, not in ansible.

File: platforms/corda-ent/configuration/roles/create/certificates/ambassador/cenm-notary/tasks/nested_main.yaml [hunk 4]
```diff
@@ -0,0 +1,229 @@
...
+    VAULT_ADDR: "{{ org.vault.url }}"
+    VAULT_TOKEN: "{{ org.vault.root_token }}"
+  register: ambassador_tls_certs_cenm
+  ignore_errors: yes
+  when: org.type == 'cenm'
+
+# Check if the ambassador tls certs are already created and stored in the vault for notary
+- name: Check if the ambassador tls certs are already created and stored in the vault for notary
+  shell: |
+    vault kv get -format=yaml secret/{{ peer.name }}/tlscerts
```
Comment (sownak) [inline platforms/corda-ent/configuration/roles/create/certificates/ambassador/cenm-notary/tasks/nested_main.yaml:121]: This is fine. But if we are separating cenm and notary, then this code needs to be under create/certificates/ambassador/notary

File: platforms/corda-ent/configuration/roles/create/certificates/ambassador/nodes/tasks/nested_main.yaml [hunk 5]
```diff
@@ -0,0 +1,203 @@
...
+  register: rootca_stat_result
+
+############################################################################################
+#################### GENERATE ROOT CA CERTIFICATES, IF ABSENT IN VAULT #####################
+
+# Generate Root CA certificates
+- name: Generate Root CA certificates
+  shell: |
+    cd {{ rootca }}
+    eval "keytool -genkey -alias rootca -dname {{ org.subject | quote }} -keystore rootca.jks -storepass quorum -keypass quorum"
```
Comment (sownak) [inline platforms/corda-ent/configuration/roles/create/certificates/ambassador/nodes/tasks/nested_main.yaml:73]: same comments as above

File: platforms/corda-ent/configuration/roles/create/certificates/ambassador/cenm-notary/tasks/nested_main.yaml (hunk 4 above)
Comment (jagpreetsinghsasan) [reply platforms/corda-ent/configuration/roles/create/certificates/ambassador/cenm-notary/tasks/nested_main.yaml:121]: We are seperating notary just for logical seperation (as an organization as we want to make that component HA in future). As per the code, I made the cenm and notary section similar.

**Pull request closed without merging**